### ISSUE TITLE
KBFS drive not visible

### ISSUE THREAD

**Issue by doxxx** [keybase/client#8676]
The KBFS K: drive is not visible. When I open the Keybase app and go to the Folders tab, it shows the warning about Keybase folders not showing up in Explorer. If I click the "Display in Explorer" link, it starts up the Keybase installer prompting me to click Repair, which I do. After the UAC prompt, it pops up a File Open dialog showing my Downloads folder with a filename like "Keybase_1.0.31-20170921162535+71e03df.386.exe" already filled in the prompt. That file does not exist in my Downloads folder. The only thing I can do is click Cancel and the installer tells me that the repair failed.

Running on Windows 10 Home version 1703.

my log id: 738efcd8fc214549d09a101c

**Comment by maxtaco** [comment]
Cc @zanderz

**Comment by doxxx** [comment]
I should also note that I've uninstalled, rebooted, reinstalled and rebooted.

**Comment by doxxx** [comment]
I just tried rerunning the installer "as administrator" and choosing the Repair option. It ran to completion without that File Open dialog behavior, but the K drive is still not visible and the same "Error when enabling existing journals" error still shows up in the kbfs log, which you should be able to see in my log.

**Comment by ThierryIT** [comment]
Are you admin on your pc ?

**Comment by zanderz** [comment]
@doxxx Sorry about these problems, we're working on making that better. Meanwhile, one thing you can do is run the originally downloaded installer with the option `DRIVER=1`, e.g. keybase_setup_386.exe DRIVER=1`

**Comment by doxxx** [comment]
@thierryit I didn’t explicitly create my user as administrator and I get the UAC prompt for installers and other admin stuff, so I would assume no.

**Comment by doxxx** [comment]
@zanderz thanks I’ll try that out

**Comment by ThierryIT** [comment]
and let us know ...

**Comment by doxxx** [comment]
Yep, didn’t get a chance to yet.

**Comment by doxxx** [comment]
I tried the original installer again with the `DRIVER=1` option and picked Repair. It ran to completion and then I rebooted as asked. When I logged in again, I just got a black screen with a cursor. I waited for a minute or two but nothing happened.

I hit Ctrl-Alt-Delete, opened the Task Manager, reran the installer and chose the Uninstall option. Almost immediately, the rest of my desktop loaded as the uninstall completed.

So... Not sure what to make of that. Anything else you would like me to try?

**Comment by ThierryIT** [comment]
I have tryed today, no change, see my picture.
![kbfs](https://user-images.githubusercontent.com/31948106/31015746-5b24656c-a52a-11e7-834d-2d8e39f20056.JPG)

**Comment by zanderz** [comment]
@doxxx Do you have any AV software installed? If you care to try again and can get to a state where you can do `keybase log send` at the Keybase Shell, we may be able to tell what happened.

@ThierryIT Is this the state where "Display in Explorer" leads to a repair install which fails because your account lacks permission? Not clear what the issue is from the picture.

**Comment by ThierryIT** [comment]
Hi,
Yes you are right, but I start to be confused concerning this lacks of permission, when we are able to install Keybase with a normal user (not admin) ...

**Comment by zanderz** [comment]
The second half of the installer includes a device driver, which requires different permissions to install. Corporately managed machines often have restrictions on what kind of things can be installed, which is an important Windows feature, not a defect. Note that Keybase is usable for chat and command line filesystem operations even without a drive being mounted (`keybase fs ...`).

**Comment by ThierryIT** [comment]
Yes that 's how I am today using it ... But this is a bit frustrating not to be able to use folders ...

**Comment by doxxx** [comment]
I just have Windows Defender enabled. I’ll try again to capture logs this evening.

**Comment by doxxx** [comment]
I just tried rerunning the installer with the DRIVER=1 parameter again. This time there was no UAC prompt and it didn't ask me to reboot to complete the installation. I uninstalled and the reran the installer as admin. Still no prompt to reboot. I rebooted anyway. Still no K drive. I collected logs -- the ID is aa350119fe9f2e2162c73f1c.

**Comment by doxxx** [comment]
I tried reinstalling after deleting the kbfs* folders in AppData/Local/Keybase, but still no dice.

**Comment by zanderz** [comment]
@doxxx Apologies, the `driver` parameter must be lower case, sorry for my mistake there.

**Comment by doxxx** [comment]
Ok, so running the installer with `driver=1` instead resulted in different status messages about the dokan library while the installer was running. There's more happening in the logs now as well, although the same two errors still appear. And there's no K drive. :)

Logs ID: 70cf41903bb24de14fbb751c

**Comment by zanderz** [comment]
It says it mounted the filesystem. Your logs also show hundreds of files being created, not sure what that's about. Did the mount never show up? No install errors either. What are the errors you see now?
Thanks again for hanging in there and for the feedback.

**Comment by doxxx** [comment]
No new drive mount appears in Windows Explorer. If I open the Keybase app (which took two tries before the window displayed), the folders page no longer shows that warning so it thinks that the filesystem is mounted. But if I click one of the folders, I get a little dropdown from the top of the window which says "openInWindows requires k prefix: C:\public\doxxx. The path it mentions does not exist.

`keybase kbfsmount status` gives me this:

```
C:\Users\Gordon\AppData\Local\Keybase>keybase kbfsmount status
- INFO CheckRegistryKeybaseDokan: Searching registry for {65A3A964-3DC3-0100-0004-170726135020}
{
  "version": "",
  "bundleVersion": "",
  "kextID": "",
  "path": "",
  "kextStarted": true,
  "installStatus": 4,
  "installAction": 1,
  "mountInfos": null,
  "status": {
    "code": 0,
    "name": "",
    "desc": "",
    "fields": null
  }
}
```

Now here's the weird part. So the K: drive is not visible or accessible in anyway in Explorer. But if I open the Keybase shell, I can type `k:` and it works. The drive exists when accessed through the shell. But if I open up a separate cmd window, no K: drive.

I just noticed that the Keybase Shell window is running as administrator, so I tried starting a separate cmd window as administrator and in there I can access the K: drive. So it appears the drive is mounted but is only visible to administrator.

**Comment by doxxx** [comment]
I just rebooted my PC again, and now the drive shows up in Explorer! It took a a minute or two to show up after I logged in. So it looks like a reboot is needed even though the installer doesn't say so.

**Comment by iloabn** [comment]
Just wanted to chip in that I had the same issue.
Running the installer with `driver=1` and then reboot solved it for me as well.

Doesn't feel like intended behavior.
